packages/pwa-kit-runtime/src/utils/ssr-server/process-lambda-response.js

```diff
@@ -6,33 +6,33 @@
  */
 
 import {CONTENT_TYPE, X_ORIGINAL_CONTENT_TYPE} from '../../ssr/server/constants'
+import {getFlattenedHeadersMap} from '@h4ad/serverless-adapter'
 
 export const processLambdaResponse = (response, event) => {
     if (!response) return response
 
     // Retrieve the correlation ID from the event headers
     const correlationId = event.headers?.['x-correlation-id']
 
-    const responseHeaders = {
-        ...response.headers
-    }
+    let joinedHeaders = getFlattenedHeadersMap(response.multiValueHeaders || {}, ',', true)
+    delete response['multiValueHeaders']
 
     // Add the correlation ID to the response headers if it exists
     if (correlationId) {
-        responseHeaders['x-correlation-id'] = correlationId
+        joinedHeaders['x-correlation-id'] = correlationId
```

Comment:
@kieran-sf I think in the future it would make sense to move the setting of the correlation response header id into app.js. What do you think?

Reply:
Yes I was just thinking we could add it to the response header in app.js as well after the express app has processed the request.